subsystem/src/manager.rs

```diff
@@ -226,13 +227,13 @@ impl Manager {
                                 worker_tasks.spawn(async move {
                                     let mut subsys = subsys.write().await;
                                     call(&mut *subsys).await
-                                });
+                                }.in_current_span());
                             },
                             Action::Ref(call) => {
                                 worker_tasks.spawn(async move {
                                     let subsys = subsys.read().await;
                                     call(&*subsys).await
-                                });
+                                }.in_current_span());
```

Comment:
I can see a future where we gradually move away from `task::spawn` in favour of various task pools and scoped tasks and similar. That means either invoking `in_current_span` manually like here or coming up with a wrapper to make it convenient (similar to the top-level `spawn_in_current_span`).

Does it make sense to address it now or should we leave it and deal with it as the need arises?

Reply:
I'm not sure how to address this now, because it's not clear to me what those `spawn` replacements will look like.
Do you have any particular suggestions?

Reply:
I'm fine doing it as the need arises and some patterns emerge